src/simulation/LoadGenerator.cpp

```diff
@@ -866,11 +1025,224 @@ LoadGenerator::manageOfferTransaction(
+
+    return std::make_pair(account, tx);
+}
+
+std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>
+LoadGenerator::invokeSorobanLoadTransaction(uint32_t ledgerNum,
+                                            uint64_t accountId,
+                                            GeneratedLoadConfig const& cfg)
+{
+    auto account = findAccount(accountId, ledgerNum);
+    auto instanceIter = mContractInstances.find(accountId);
+    releaseAssert(instanceIter != mContractInstances.end());
+    auto const& instance = instanceIter->second;
+
+    auto const& networkCfg = mApp.getLedgerManager().getSorobanNetworkConfig();
+
+    // Approximate instruction measurements from loadgen contract. While the
+    // guest and host cycle counts are accurate, it very difficult to estimate
+    // the number of instructions that the write loops will consume
+    // without preflight, so some TXs will fail due to exceeding resource
+    // limitations. However these should fail at apply time, so will still
+    // generate siginificant load
+    uint64_t const baseInstructionCount = 3'000'000;
+    uint64_t const instructionsPerGuestCycle = 120;
+    uint64_t const instructionsPerHostCycle = 2355;
+
+    // Pick random number of cycles between bounds, respecting network limits
+    uint64_t maxInstructions =
+        networkCfg.mTxMaxInstructions - baseInstructionCount;
+
+    uint64_t guestCyclesMax = maxInstructions / instructionsPerGuestCycle;
+    guestCyclesMax = std::min(cfg.guestCyclesHigh, guestCyclesMax);
+    uint64_t guestCyclesMin = std::min(cfg.guestCyclesLow, guestCyclesMax);
+    uint64_t guestCycles =
+        rand_uniform<uint64_t>(guestCyclesMin, guestCyclesMax);
+
+    maxInstructions -= guestCycles * instructionsPerGuestCycle;
+    uint64_t hostCyclesMax = maxInstructions / instructionsPerHostCycle;
+    hostCyclesMax = std::min(cfg.hostCyclesHigh, hostCyclesMax);
+    uint64_t hostCyclesMin = std::min(cfg.hostCyclesLow, hostCyclesMax);
+    uint64_t hostCycles = rand_uniform<uint64_t>(hostCyclesMin, hostCyclesMax);
+
+    SorobanResources resources;
+    resources.footprint.readOnly = instance.readOnlyKeys;
+
+    // Must always read wasm and instance
+    releaseAssert(networkCfg.mTxMaxReadLedgerEntries > 1);
+    auto maxEntries = networkCfg.mTxMaxReadLedgerEntries - 2;
+    maxEntries = std::min(maxEntries, cfg.nDataEntriesHigh);
+    auto minEntries = std::min(maxEntries, cfg.nDataEntriesLow);
+    auto numEntries = rand_uniform<uint32_t>(minEntries, maxEntries);
+    for (uint32_t i = 0; i < numEntries; ++i)
+    {
+        auto lk = contractDataKey(instance.contractID, makeU32(i),
+                                  ContractDataDurability::PERSISTENT);
+        resources.footprint.readWrite.emplace_back(lk);
+    }
+
+    // Use 2'000 as an estimate for Wasm size
+    uint32_t maxKiloBytesPerEntry =
+        (networkCfg.mTxMaxReadBytes - 2'000) / numEntries / 1024;
```

Comment:
nit: It should actually be easy to just get the actual Wasm size (which is just the length of the Wasm bytes vector).